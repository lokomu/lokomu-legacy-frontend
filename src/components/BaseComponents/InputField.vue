<template>
  <input
    :id="id"
    class="w-full text-center text-sm p-2.5 outline-none rounded-lg"
    :class="validationClasses"
    :type="type"
    :placeholder="placeholder"
    :value="modelValue"
    @input="onInput"
    @keyup.enter="onEnter"
  >
</template>

<script setup lang="ts">
import { computed } from "vue";

const props = withDefaults(
  defineProps<{
    type: string;
    id: string;
    placeholder: string;
    modelValue: string;
    isValid: boolean | null;
  }>(),
  {
    type: "text",
    isValid: null,
  },
);

const baseClasses =
  "bg-gray-100 dark:bg-gray-700 dark:text-gray-400 dark:placeholder-gray-500 focus:border-green-500 focus:ring-2 focus:ring-green-500 focus:ring-opacity-50 dark:focus:border-green-500";
const validClasses =
  "bg-green-50 border border-green-500 text-green-900 dark:text-green-400 placeholder-green-700 dark:placeholder-green-500 dark:bg-gray-700 dark:border-green-500 focus:ring-green-500 focus:border-green-500";
const invalidClasses =
  "bg-red-50 border border-red-500 text-red-900 placeholder-red-700 text-sm dark:bg-gray-700 focus:border-red-500 dark:text-red-500 dark:placeholder-red-500 dark:border-red-500";

const validationClasses = computed(() => {
  return props.isValid === null
    ? baseClasses
    : props.isValid
    ? validClasses
    : invalidClasses;
});

const emit = defineEmits(["update:modelValue", "enter"]);

const onInput = (event: Event) => {
  if (event.target) {
    emit("update:modelValue", (event.target as HTMLInputElement).value);
  }
};
const onEnter = () => emit("enter");
</script>
