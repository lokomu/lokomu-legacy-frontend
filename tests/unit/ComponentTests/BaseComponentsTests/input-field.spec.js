import { mount } from '@vue/test-utils';
import InputField from '@/components/BaseComponents/InputField.vue';
import {describe, it, expect} from "vitest";

describe('InputField', () => {

    const wrapper = mount(InputField, {
        props: {
            modelValue: '',
            isValid: null,
        },
    });

    it('applies the correct classes based on isValid prop', async () => {
        // check that the base classes are applied when isValid is null
        expect(wrapper.classes()).toContain('bg-gray-100');

        await wrapper.setProps({ isValid: true });
        expect(wrapper.classes()).toContain('bg-green-50');

        await wrapper.setProps({ isValid: false });
        expect(wrapper.classes()).toContain('bg-red-50');
    });

    it("renders correctly", () => {
        expect(wrapper.element).toMatchSnapshot();
    });

    it("is instantiated", () => {
        expect(wrapper.exists()).toBeTruthy();
    });
});