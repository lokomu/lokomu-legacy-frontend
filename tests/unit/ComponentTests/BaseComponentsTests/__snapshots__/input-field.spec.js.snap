// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputField > renders correctly 1`] = `
<input
  class="w-full text-center text-sm p-2.5 outline-none rounded-lg bg-red-50 border border-red-500 text-red-900 placeholder-red-700 text-sm dark:bg-gray-700 focus:border-red-500 dark:text-red-500 dark:placeholder-red-500 dark:border-red-500"
  type="text"
/>
`;
